testbed/tests/app/test_desktop.py

```diff
@@ -354,73 +361,74 @@ async def test_presentation_mode_exit_on_window_state_change(
     window2.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
     window1.show()
     window2.show()
-    # Add delay to ensure windows are visible after animation.
+    # Wait for window animation before assertion.
     await main_window_probe.wait_for_window("Test windows are shown")
+
     # Enter presentation mode
     app.enter_presentation_mode([window1])
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
-        "App is in presentation mode", full_screen=True
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
+        "App is in presentation mode", state=WindowState.PRESENTATION
     )
 
     assert app.in_presentation_mode
     assert window1_probe.instantaneous_state == WindowState.PRESENTATION
 
     # Changing window state of main window should make the app exit presentation mode.
     window1.state = new_window_state
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
         "App is not in presentation mode" f"\nTest Window 1 is in {new_window_state}",
-        minimize=True if new_window_state == WindowState.MINIMIZED else False,
-        full_screen=True if new_window_state == WindowState.FULLSCREEN else False,
+        state=new_window_state,
     )
 
     assert not app.in_presentation_mode
     assert window1_probe.instantaneous_state == new_window_state
 
     # Reset window states
     window1.state = WindowState.NORMAL
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
+        "All test windows are in WindowState.NORMAL", state=WindowState.NORMAL
+    )
```

Comment:
This assertion should be after the window2.state assignment, so that both windows can be restored in parallel, rather than running strictly in series. That will make the test marginally faster.

Reply:
Thanks, I have changed it.